settlement/dymension/dymension.go

```diff
@@ -502,6 +503,7 @@ func (c *Client) convertBatchToMsgUpdateState(batch *types.Batch, daResult *da.R
 		DAPath:      daResult.SubmitMetaData.ToPath(),
 		BDs:         rollapptypes.BlockDescriptors{BD: blockDescriptors},
 		Last:        batch.LastBatch,
+		DrsVersion:  version.Commit,
```

Comment:
@keruch where is this value validated across the L2?
what prevents the sequencer from lying?

Reply:
I see that each node validates his own version vs the rollapp params
```
if version.Commit != m.State.RollappParams.Version {
    return fmt.Errorf("binary version mismatch. rollapp param: %s binary used:%s", m.State.RollappParams.Version, version.Commit)
}
```

but I don't see where it's validated in regard with the state updates.

let's assume L2 runs version 8.
but the sequencer sets `DrsVersion: 9` in the state update (e.g to bypass the DRS check)

Reply:
@danwt as discussed, this scenario should be fraud provable
pls link the related research/ADR/issue